chain/client/src/chunk_producer.rs

```diff
@@ -279,12 +280,16 @@ impl ChunkProducer {
         )?;
 
         let outgoing_receipts_root = self.calculate_receipts_root(epoch_id, &outgoing_receipts)?;
-        let protocol_version = self.epoch_manager.get_epoch_protocol_version(epoch_id)?;
         let gas_used = chunk_extra.gas_used();
         #[cfg(feature = "test_features")]
         let gas_used = if self.produce_invalid_chunks { gas_used + 1 } else { gas_used };
 
         let congestion_info = chunk_extra.congestion_info();
+        let bandwidth_requests = chunk_extra.bandwidth_requests();
```

Comment:
does it make sense for all the `bandwidth_requests()` functions to stop returning an `Option`.  It seems like all the callers of this function are now `unwrap`ing the response.

Reply:
We could do that by introducing a panic when `bandwidth_requests` is called on `Self::V1(_) | Self::V2(_) | Self::V3(_)`, similar to what we have for `congestion_info_mut` [here](https://github.com/darioush/nearcore/blob/dc8154d9aade83502334ce8a1e00b51c6b24870a/core/primitives/src/types.rs#L931)

There was some discussion about whether to remove it or not [here](https://github.com/near/nearcore/pull/13402#discussion_r2053277741).

Happy to make the modification if you think it is an improvement.

Reply:
Looking at the way you handle the `None` case in the current PR, in each case, you substitute `BandwidthRequests::empty`.  So I was thinking that you update `bandwidth_request()` to return `BandwidthRequests::empty` in the cases where you are returning `None` currently.  No extra panics needed.

As an aside, I think it might be risky to panic for older versions.  In the case of archival nodes, those cases might actually be hit and we wouldn't want to panic there.  And it might also be possible for malicious nodes to send us older versions of structs which we currently generally accept so that can also lead to potential panics.

Reply:
This would require changing the return type to `BandwidthRequests` instead of `&BandwidthRequests`, I tried that here: https://github.com/darioush/nearcore/pull/1/files if you prefer it you can merge it to this PR.

(adds a clone [here](https://github.com/darioush/nearcore/pull/1/files#diff-f3b6508a21adda04d4f9cdf1d08be95f3df9307ad88e7f81caa15de402629ad2R138), maybe fine)

Reply:
Ah now I get the complexity. Sorry it took me a while